### PR TITLE
Disable connection caching for connections 

### PR DESCRIPTION
**Pull request by tli-ocient**
See comment added in code. This isn't a performance concern since that particular flow requires human interaction.

### CODE REVIEW

File: src/main/java/com/ocient/jdbc/XGConnection.java [hunk 1]
```diff
@@ -2196,23 +2197,37 @@ public SQLWarning getWarnings() throws SQLException
 		return retval;
 	}
 
+	@Override
+	public boolean equals(Object o) {
```
Comment (jwilliams-ocient) [inline src/main/java/com/ocient/jdbc/XGConnection.java:2201]: Add a comment saying `hashCode() & equals()` should almost ALWAYS be autogenerated, and to only roll your own if you:
1. have a legitimate reason to do so (v rare) 
2. know what you're doing
3.  write unit tests specifically for object equality

File: userdoc/release_notes.adoc [hunk 2]
```diff
@@ -11,6 +11,7 @@ Below is the release notes for every driver version that has customer implicatio
 New Features::
 
  * Data extract tool released.
+ * Deactive connection caching for SSO flow.
```
Comment (jwilliams-ocient) [inline userdoc/release_notes.adoc:14]: Maybe specify the caching is disabled for calls to `Driver.createConnection(...)` (as opposed to `Connection.createStatement(...)`)

File: src/main/java/com/ocient/jdbc/XGConnection.java [hunk 3]
```diff
@@ -1794,18 +1795,18 @@ public Struct createStruct(final String arg0, final Object[] arg1) throws SQLExc
...
 
-	@Override
-	public boolean equals(final Object o)
-	{
-		if (!(o instanceof XGConnection))
-		{
-			return false;
-		}
+	// @Override
+	// public boolean equals(final Object o)
```
Comment (jwilliams-ocient) [inline src/main/java/com/ocient/jdbc/XGConnection.java:1799]: delete comments

File: src/main/java/com/ocient/jdbc/JDBCDriver.java [hunk 4]
```diff
@@ -291,9 +291,11 @@ public Connection createConnection(final String hostname, final int portNum, fin
 						conn.connect();
 						LOGGER.log(Level.INFO, "Successfully connected");
 						connected = true;
-						synchronized (seenConnections)
-						{
-							seenConnections.put(conn, conn);
+						if(!conn.isDisabledConnCaching()){
```
Comment (jwilliams-ocient) [inline src/main/java/com/ocient/jdbc/JDBCDriver.java:294]: this check is redundant?

Comment (jwilliams-ocient) [reply src/main/java/com/ocient/jdbc/JDBCDriver.java:294]: hmm, maybe not. but something is fishy here...

Comment (jwilliams-ocient) [reply src/main/java/com/ocient/jdbc/JDBCDriver.java:294]: yea, I think it is redundant.

1. conn_exist  ❌     |    caching  ❌      => `do_connect = true`
2. conn_exist  ✅    |    caching  ❌       => `do_connect = true`
3. conn_exist  ❌    |    caching  ✅       => `do_connect = true`
4. conn_exist   ✅   |    caching  ✅       => `do_connect = false`

Comment (jwilliams-ocient) [reply src/main/java/com/ocient/jdbc/JDBCDriver.java:294]: Do (1) and (2) reduce to the same case? Maybe that's my confusion... Like (2) should _never_ happen

Comment (tli-ocient) [reply src/main/java/com/ocient/jdbc/JDBCDriver.java:294]: Hmm, if no connection exists and caching is disabled (case 1), then do_connect is true and we add this to the cache(without that second check)

Comment (tli-ocient) [reply src/main/java/com/ocient/jdbc/JDBCDriver.java:294]: then I think 2 will happen?

Comment (jwilliams-ocient) [reply src/main/java/com/ocient/jdbc/JDBCDriver.java:294]: I guess that's where I'm confused. What other pieces of code update `seenConnections`?

Comment (tli-ocient) [reply src/main/java/com/ocient/jdbc/JDBCDriver.java:294]: nowhere else, just here

Comment (jwilliams-ocient) [reply src/main/java/com/ocient/jdbc/JDBCDriver.java:294]: and is there a problem if that happens w.r.t. the cache in `XGStatement`. 

RE `hashCode()` & `equals(...)` for `XGConnection`.

Comment (tli-ocient) [reply src/main/java/com/ocient/jdbc/JDBCDriver.java:294]: I don't think so. I'm just concerned about us adding connections unecessarily to the `seenConnections` cache

Comment (jwilliams-ocient) [reply src/main/java/com/ocient/jdbc/JDBCDriver.java:294]: why? Not sure I understand that. AFAICT we are strictly "reducing" the number of elements in `seenConnections`?

Comment (tli-ocient) [reply src/main/java/com/ocient/jdbc/JDBCDriver.java:294]: Thats not true, we aren't removing any connections from here, ever. Only adding

Comment (jwilliams-ocient) [reply src/main/java/com/ocient/jdbc/JDBCDriver.java:294]: Also, the statement pooling in `XGStatement` would _still_ broken in SSO contexts...

File: userdoc/release_notes.adoc (hunk 2 above)
Comment (tli-ocient) [reply userdoc/release_notes.adoc]: changed comment

File: src/main/java/com/ocient/jdbc/XGConnection.java (hunk 3 above)
Comment (tli-ocient) [reply src/main/java/com/ocient/jdbc/XGConnection.java]: done

File: src/main/java/com/ocient/jdbc/JDBCDriver.java (hunk 4 above)
Comment (tli-ocient) [reply src/main/java/com/ocient/jdbc/JDBCDriver.java:294]: talked in zoom. this was necessary

**Pull request merged**